include/pybind11/pytypes.h

```diff
@@ -1336,9 +1337,14 @@ class memoryview : public object {
         // Py_buffer uses signed sizes, strides and shape!..
         static std::vector<Py_ssize_t> py_strides { };
         static std::vector<Py_ssize_t> py_shape { };
+        static std::vector<std::string> formats = {
+            "?", "b", "B", "h", "H", "i", "I", "q", "Q", "f", "d", "g"
+        };
```

Comment:
Hm, wondering if this could be just `static char* format` given they are all length-1?

Reply:
Ah, apparently it can be length 2 as noted above; well, could probably use something like a `static const char* format[]`?